integration/src/main/java/scripts/utils/CloudContextMaker.java

```diff
@@ -29,14 +30,47 @@ public static String createGcpCloudContext(UUID workspaceUuid, WorkspaceApi work
             .jobControl(new JobControl().id(contextJobId));
 
     logger.info("Creating GCP cloud context");
+    CreateCloudContextResult contextResult =
+        waitForCloudContext(createContext, workspaceUuid, workspaceApi, contextJobId);
+    return contextResult.getGcpContext().getProjectId();
+  }
+
+  /**
+   * Creates an Azure cloud context for a given workspace. Returns the Azure resource group ID as a
+   * string
+   */
+  public static String createAzureCloudContext(
+      UUID workspaceUuid, WorkspaceApi workspaceApi, AzureContext context) throws Exception {
+    String contextJobId = UUID.randomUUID().toString();
+
+    var createContext =
+        new CreateCloudContextRequest()
+            .cloudPlatform(CloudPlatform.AZURE)
+            .jobControl(new JobControl().id(contextJobId))
+            .azureContext(context);
+
+    logger.info("Creating Azure cloud context");
+    CreateCloudContextResult contextResult =
+        waitForCloudContext(createContext, workspaceUuid, workspaceApi, contextJobId);
+    return contextResult.getAzureContext().getResourceGroupId();
+  }
+
+  private static CreateCloudContextResult waitForCloudContext(
+      CreateCloudContextRequest createContext,
+      UUID workspaceUuid,
+      WorkspaceApi workspaceApi,
+      String contextJobId)
+      throws Exception {
     CreateCloudContextResult contextResult =
         workspaceApi.createCloudContext(createContext, workspaceUuid);
     while (ClientTestUtils.jobIsRunning(contextResult.getJobReport())) {
       Thread.sleep(CREATE_CONTEXT_POLL_INTERVAL.toMillis());
       contextResult = workspaceApi.getCreateCloudContextResult(workspaceUuid, contextJobId);
     }
     logger.info(
-        "Create GCP context status is {}", contextResult.getJobReport().getStatus().toString());
+        "Create {} context status is {}",
+        createContext.getCloudPlatform().equals(CloudPlatform.GCP) ? "GCP" : "Azure",
```

Comment:
nit: you could also use `createContext.getCloudPlatform().toSql()` to get a readable string, either way is fine